build.sbt

```diff
@@ -57,13 +57,15 @@ lazy val runtime2         = Project("daffodil-runtime2", file("daffodil-runtime2
                                 Compile / ccTargets := ListSet(runtime2CFiles),
                                 Compile / cSources  := Map(
                                   runtime2CFiles -> (
-                                    ((Compile / resourceDirectory).value / "c" * GlobFilter("*.c")).get() ++
+                                    ((Compile / resourceDirectory).value / "c" / "libcli" * GlobFilter("*.c")).get() ++
+                                    ((Compile / resourceDirectory).value / "c" / "libruntime" * GlobFilter("*.c")).get() ++
```

Comment:
Just an FYI, you can use ** to get a recusrive glob, for example, you could instead do something like:
```scala
((Compile / resourceDirectory).value / "c" / ** GlobFilter("*.c")).get()
```
Though, being explicit might have some advantages?

Another question, since the libcli and libruntime directories are both added to runtime2CFiles, does that mean both libs end up in the same .so/.a files? Same goes for the examples file? If we're separating out source, do we also want to separate the resulting compiled objects? Or is this purley organizational?

Another related question, I see we are also compiling the examples files into this same .a/so file? Should that be separate as well? And chance of conflicts between examples, libcli, and libruntime?